ADR/0006-log-conventions.md

```diff
@@ -11,39 +12,36 @@ Accepted
 
 We need log conventions to make our offering easier to operate and maintain. These will also make it possible for us to create queries that cross components in our logs, which supports [STONE-201](https://issues.redhat.com/browse/STONE-201).
 
-## Decision: Log conventions
+We initially decided to require log messages formatted by the `console` encoder, thinking that it would be advantageous to be able to consume logs in a human readable format directly from our controller pods (2023-03-08), but on [subsequent review](https://redhat-internal.slack.com/archives/C02C3SE8QS2/p1680074942662149?thread_ts=1680071345.179399&cid=C02C3SE8QS2) we realized that this bears an undesirable cost in configuration of our logging: the default encoder for the "production" profile in zap is the `json` encoder (2023-03-29).
 
-In our controller logs, we will use structured log messages, partially formatted in JSON with key-value pairs as described below.
+## Decision: Log conventions
 
-```
-TIMESTAMP LOG_LEVEL LOGGER MESSAGE ADDITIONAL_JSON_DATA
-```
+In our controller logs, we will use structured log messages, formatted in JSON with key-value pairs as described below.
```

Comment:
Log messages in JSON format might be good for production use-case, but for development it's not that great, because the information a developer want to see first gets scattered in the JSON.
In the default logging format (generated by operator-sdk), log messages looks like:
```
2023-02-17T10:41:10Z	ERROR	controllers.ComponentOnboarding	failed to update initial build annotation	{"ComponentOnboarding": "integ-e2e-qplx/integration-suite-test-component-git-source-hwtj", "error": "Operation cannot be fulfilled on components.appstudio.redhat.com \"integration-suite-test-component-git-source-hwtj\": the object has been modified; please apply your changes to the latest version and try again"}
```
so it's easy to find any errors occurred and the short error description is also at the beginning. If a developer needs more details on the error, it's easy to read the line further.

In the proposed JSON format, can we guarantee the fields order, so it's easy to read the log?